packages/mui-joy/src/Button/Button.tsx

```diff
@@ -1,21 +1,258 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_capitalize as capitalize, unstable_useForkRef as useForkRef } from '@mui/utils';
+import { useButton } from '@mui/base/ButtonUnstyled';
+import composeClasses from '@mui/base/composeClasses';
+import { styled, useThemeProps } from '../styles';
+import { rootShouldForwardProp } from '../styles/styled';
+import { ExtendButton, ButtonTypeMap, ButtonProps } from './ButtonProps';
+import buttonClasses, { getButtonUtilityClass } from './buttonClasses';
 
-export type ButtonProps = JSX.IntrinsicElements['button'];
+const useUtilityClasses = (ownerState: ButtonProps & { focusVisible: boolean }) => {
+  const {
+    classes,
```

Comment:
Do we really need the `classes`. We have them in v5 so that we don't have many breaking changes. You can still apply classes for the slots inside `componentsProps[slot].className`.

Reply:
Agree with @mnajdova. I decided not to use classes in Base components anymore.

Reply:
`classes` prop is removed.

Reply:
> Do we really need the `classes` prop?

We will see. Trying without sounds like a great idea to see how developers react. I imagine that some could still be using them, especially with SASS and CSS modules, but maybe they will be OK with the alternatives.